cvmfs/catalog_diff_tool_impl.h

```diff
@@ -178,8 +189,9 @@ void CatalogDiffTool<RoCatalogMgr>::DiffRec(const PathString& path) {
       if (id_nested_from == id_nested_to) continue;
     }
 
-    if ((diff != catalog::DirectoryEntryBase::Difference::kIdentical) ||
-        old_entry.IsNestedCatalogMountpoint()) {
+    if (IsReportablePath(old_path) &&
```

Comment:
Not sure about this condition: If we find a difference between old and new on a non-reportable path, we won't recurse further into the tree.  Is this the desired logic?

Reply:
No, that `if` statement guards only the call to `ReportModification()`.  If `IsReportablePath()` returns false then the call to `ReportModification()` will be skipped (along with the potential early termination that `ReportModification()` may request), but the control flow will continue to the normal recursion path that follows the `if` block.

I'll make sure there's a test case that covers this case.

Reply:
I've extended the integration tests to include situations where a path directly above the lease path (i.e. neither ignored nor reportable) is modified, and confirmed that the behaviour is correct.